packages/material/src/controls/MaterialSimpleAnyOfControl.tsx

```diff
@@ -0,0 +1,139 @@
+
+  The above copyright notice and this permission notice shall be included in
+  all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+import {
+  and,
+  ControlProps,
+  ControlState,
+  EnumCellProps,
+  JsonSchema,
+  mapDispatchToControlProps,
+  mapStateToControlProps,
+  RankedTester,
+  rankWith,
+  schemaMatches,
+  uiTypeIs,
+  WithClassname
+} from '@jsonforms/core';
+import { Control } from '@jsonforms/react';
+import { Input } from '@material-ui/core';
+import { InputBaseComponentProps } from '@material-ui/core/InputBase';
+import merge from 'lodash/merge';
+import React from 'react';
+import { connect } from 'react-redux';
+import { MaterialInputControl } from './MaterialInputControl';
+
+const MuiAutocompleteInputText = (props: EnumCellProps & WithClassname) => {
+  const {
+    data,
+    config,
+    className,
+    id,
+    enabled,
+    uischema,
+    isValid,
+    path,
+    handleChange,
+    schema
+  } = props;
+
+  const findEnumSchema = (schemas: JsonSchema[]) =>
+    schemas.find(
+      s => s.enum !== undefined && (s.type === 'string' || s.type === undefined)
+    );
+  const findTextSchema = (schemas: JsonSchema[]) =>
+    schemas.find(s => s.type === 'string' && s.enum === undefined);
+  const enumSchema = findEnumSchema(schema.anyOf);
+  const textSchema = findTextSchema(schema.anyOf);
+  const maxLength = textSchema.maxLength;
+  const mergedConfig = merge({}, config, uischema.options);
+  let inputProps: InputBaseComponentProps = {};
+  if (mergedConfig.restrict) {
+    inputProps = { maxLength: maxLength };
+  }
+  if (mergedConfig.trim && maxLength !== undefined) {
+    inputProps.size = maxLength;
+  }
+  const onChange = (ev: any) => handleChange(path, ev.target.value);
+
+  inputProps.list = props.id + 'datalist';
+  const dataList = (
+    <datalist id={props.id + 'datalist'}>
+      {enumSchema.enum.map(optionValue => (
+        <option value={optionValue} key={optionValue} />
+      ))}
+    </datalist>
+  );
+  return (
+    <Input
+      type='text'
+      value={data || ''}
+      onChange={onChange}
+      className={className}
+      id={id}
+      disabled={!enabled}
+      autoFocus={uischema.options && uischema.options.focus}
+      fullWidth={!mergedConfig.trim || maxLength === undefined}
+      inputProps={inputProps}
+      error={!isValid}
+      endAdornment={dataList}
+    />
+  );
+};
+
+export class MaterialSimpleAnyOfControl extends Control<ControlProps, ControlState> {
```

Comment:
SimpleAnyOf is a somewhat misleading name, as this renderer is specific to type string with enums.